lib/index.js

```diff
@@ -631,20 +641,72 @@ function Stream(generator) {
 }
 inherits(Stream, EventEmitter);
 
-/**
- * adds a top-level _.foo(mystream) style export for Stream methods
- */
+function _addMethod(proto, topLevel) {
+    return function(name, f, options) {
+        proto[name] = f;
+        var n = f.length;
+        if (options && options.expose) {
+            function relevel(coerce) {
+                return _.ncurry(n + 1, function () {
+                    var args = slice.call(arguments);
+                    var s = coerce(args.pop());
+                    return f.apply(s, args);
+                });
+            }
+            topLevel[name] = relevel(topLevel);
+            topLevel[name]._relevel = relevel;
+        }
+    };
+}
 
-function exposeMethod(name) {
-    var f = Stream.prototype[name];
-    var n = f.length;
-    _[name] = _.ncurry(n + 1, function () {
-        var args = slice.call(arguments);
-        var s = _(args.pop());
-        return f.apply(s, args);
-    });
+var addMethod = _addMethod(Stream.prototype, _);
+
+function _addMethods(proto, topLevel, methods) {
+    for (var p in methods) {
+        if (hasOwn.call(methods, p)) {
+            _addMethod(proto, topLevel)(p, methods[p], {expose: true});
+        }
+    }
+}
+
+function addToplevelMethod(name, fn) {
+    function relevel(topLevel) {
+        var bound = fn.bind(topLevel);
+        bound._relevel = relevel;
+        return bound;
+    }
+    _[name] = relevel(_);
+}
+
+function use(Super, originalTopLevel) {
+    return function(methods) {
+        function Sub() {
+            Stream.apply(this, arguments);
+        }
+        inherits(Sub, Super);
+
+        function topLevel() {
+            return __(Sub).apply(null, arguments);
+        }
+        for (var p in originalTopLevel) {
+            if (hasOwn.call(originalTopLevel, p)) {
+                var fn = originalTopLevel[p];
+                topLevel[p] = (typeof fn._relevel === 'function') ? fn._relevel(topLevel) : fn;
+            }
+        }
+        _addMethods(Sub.prototype, topLevel, methods || {});
+        topLevel.use = use(Sub, topLevel);
+
+        return topLevel;
+    };
 }
 
+_.use = use(Stream, _);
+
+addMethod('create', function() {
+    return __(this.constructor).apply(null, arguments);
```

Comment:
Is it worth it to have `addMethod('create')` instead of just setting `prototype.create` manually? This creates a new function object every time only to use it once before throwing it away.